components/AudioPlayer.tsx

```diff
@@ -0,0 +1,388 @@
+import { Box, IconButton } from "@mui/material";
+import { useCallback, useEffect, useRef, useState } from "react";
+
+const AudioPlayer = () => {
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [duration, setDuration] = useState(0);
+  const [currentTime, setCurrentTime] = useState(0);
+  const [activeVolume, setActiveVolume] = useState(false);
+  const [isFirstRender, setIsFirstRender] = useState(true);
+  const [volume, setVolume] = useState(100);
+
+  const audioPlayer = useRef<HTMLAudioElement>(null);
+  const progressBar = useRef<HTMLInputElement>(null);
+  const volumeBar = useRef<HTMLInputElement>(null);
+
+  const pageInitialized = useRef(false);
+
+  const animationRef = useRef(0);
+  const timeoutRef = useRef<NodeJS.Timeout>();
+
+  const isMobile = () => {
```

Comment:
It does not have to be a function, make it a boolean, because this value does not really change